### ISSUE TITLE
pipenv for package versioning?

### ISSUE THREAD

**Issue by kratsg** [diana-hep/pyhf#187]
# Description

Should we use `pipenv` to manage the installed package versions?

# Relevant Issues and Pull Requests

This came up as a result of #186. 

# Bug Report

Not a bug report.

**Comment by matthewfeickert** [comment]
I like and use pipenv, but does it make sense to here? I guess this comes down to the question of do we view pyhf as [a library (`setup.py`) or an application (`Pipfile`)](https://docs.pipenv.org/advanced/#pipfile-vs-setuppy)?

In my mind we are a library so I don't think I see the advantage for using pipenv here. Can you guys detail your thoughts?

**Comment by lukasheinrich** [comment]
yeah I wouldn't add a Pipfile.. I was mostly thinking in the development docs (where we recommend to use venvs) we should use pipenv

**Comment by kratsg** [comment]
Yeah, I'm not sure if we need this. Do we expect a lot of developers and headache for maintenance? It's a nice feature, but it's also an extra thing.

**Comment by matthewfeickert** [comment]
>Do we expect a lot of developers and headache for maintenance? 

I think it is just going to be the 3 of us for the (at least) near future.

>It's a nice feature, but it's also an extra thing.

I really don't have strong feelings on this one way or another. We wouldn't need to maintain anything beyond the `setup.py` that we already have though, as I think that we can just state that any developers would create a `Pipfile.lock` and [install dependencies with](https://pipenv.readthedocs.io/en/latest/advanced/#pipfile-vs-setuppy)

```
pipenv install -e .[develop,tensorflow,mxnet,torch]
```

I do _really_ like pipenv, but @lukasheinrich the motivation in your mind is that the `Pipfile.lock` will give the developers a more consistent environment in which they're all developing in? Or am I misunderstanding?

**Comment by lukasheinrich** [comment]
I think I might have confused us with this suggestion. I only meant gently suggesting using pipenv in the `CONTRIBUTING.md` and not adding either a Pipfile nor a Pipfile.lock.

**Comment by matthewfeickert** [comment]
Okay, maybe to just tie this up we could say

> To develop, we suggest using [virtual environments](https://virtualenvwrapper.readthedocs.io/en/latest/) together with  `pip` or using [pipenv](https://pipenv.readthedocs.io/en/latest/).

?

It gets the same message across, but also points towards an incredibly useful tool. Thoughts?

**Comment by lukasheinrich** [comment]
yeah looks good.

**Comment by kratsg** [comment]
LGTM
